src/builders.ts

```diff
@@ -132,6 +133,35 @@ export function broadcastTransaction(transaction: StacksTransaction, network: St
   });
 }
 
+/**
+ * Fetch a contract's ABI
+ *
+ * @param {string} address - the contracts address
+ * @param {string} contractName - the contracts name
+ * @param {Stacks} network - the Stacks network to broadcast transaction to
+ *
+ * @returns {Promise} that resolves to a ClarityAbi if the operation succeeds
+ */
+export async function getAbi(
+  address: string,
+  contractName: string,
+  network: StacksNetwork
+): Promise<ClarityAbi> {
+  const options = {
+    method: 'GET',
+  };
+
+  const url = network.getAbiApiUrl(address, contractName);
+
+  return fetchPrivate(url, options).then(async response => {
+    if (response.ok) {
+      return JSON.parse(await response.text());
+    } else {
+      return response.text();
```

Comment:
Doesn't look like error responses will be handled correctly here. 

I usually use a pattern like this to create good fetch error objects:
```ts
    const result = await fetch(...);
    if (!result.ok) {
      let msg = '';
      try {
        // try to get the error response body which often contains a useful message
        msg = await result.text();
      } catch (error) {
        // ignore error reading response (happens if the fetch error is a result of a network issue)
      }
      // include the http status code which also tends to be useful
      throw new Error(`Response ${result.status}: ${result.statusText} fetching ${url} - ${msg}`);
    }
```